src/wp-includes/block-template-utils.php

```diff
@@ -1,23 +1,481 @@
 <?php
 /**
- * Utilities used to fetch and create templates.
+ * Utilities used to fetch and create templates and template parts.
  *
  * @package WordPress
  * @since 5.8.0
  */
 
+// Define constants for supported wp_template_part_area taxonomy.
+if ( ! defined( 'WP_TEMPLATE_PART_AREA_HEADER' ) ) {
+	define( 'WP_TEMPLATE_PART_AREA_HEADER', 'header' );
+}
+if ( ! defined( 'WP_TEMPLATE_PART_AREA_FOOTER' ) ) {
+	define( 'WP_TEMPLATE_PART_AREA_FOOTER', 'footer' );
+}
+if ( ! defined( 'WP_TEMPLATE_PART_AREA_SIDEBAR' ) ) {
+	define( 'WP_TEMPLATE_PART_AREA_SIDEBAR', 'sidebar' );
+}
+if ( ! defined( 'WP_TEMPLATE_PART_AREA_UNCATEGORIZED' ) ) {
+	define( 'WP_TEMPLATE_PART_AREA_UNCATEGORIZED', 'uncategorized' );
+}
+
+/**
+ * Returns a filtered list of allowed area values for template parts.
+ *
+ * @since 5.9.0
+ *
+ * @return array The supported template part area values.
+ */
+function get_allowed_block_template_part_areas() {
+	$default_area_definitions = array(
+		array(
+			'area'        => WP_TEMPLATE_PART_AREA_UNCATEGORIZED,
+			'label'       => __( 'General' ),
+			'description' => __(
+				'General templates often perform a specific role like displaying post content, and are not tied to any particular area.'
+			),
+			'icon'        => 'layout',
+			'area_tag'    => 'div',
+		),
+		array(
+			'area'        => WP_TEMPLATE_PART_AREA_HEADER,
+			'label'       => __( 'Header' ),
+			'description' => __(
+				'The Header template defines a page area that typically contains a title, logo, and main navigation.'
+			),
+			'icon'        => 'header',
+			'area_tag'    => 'header',
+		),
+		array(
+			'area'        => WP_TEMPLATE_PART_AREA_FOOTER,
+			'label'       => __( 'Footer' ),
+			'description' => __(
+				'The Footer template defines a page area that typically contains site credits, social links, or any other combination of blocks.'
+			),
+			'icon'        => 'footer',
+			'area_tag'    => 'footer',
+		),
+	);
+
+	/**
+	 * Filters the list of allowed template part area values.
+	 *
+	 * @param array $default_areas An array of supported area objects.
+	 */
```

Comment:
Filters also need an `@since`.
```suggestion
	/**
	 * Filters the list of allowed template part area values.
	 *
	 * @since 5.9.0
	 *
	 * @param array $default_areas An array of supported area objects.
	 */
```